pycbc/workflow/pegasus_sites.py

```diff
@@ -223,7 +224,20 @@ def add_osg_site(sitecat, cp):
 
 def add_site(sitecat, sitename, cp, out_dir=None):
     """Add site sitename to sitecatalog"""
-    if out_dir is None:
+    # Allow local site scratch to be overriden for any site which uses it
+    sec = 'pegasus_profile-{}'.format(sitename)
+    opt = 'pycbc|site-scratch'
+    if cp.has_option(sec, opt):
+        out_dir = os.path.abspath(cp.get(sec, opt))
+        if cp.has_option(sec, 'pycbc|unique-scratch'):
```

Comment:
What is the motivation for having this? The scratch directories *are* (and must be) unique anyway, they are, before this patch, just stored as unique subdirectories within the same site-scratch directory, and I don't think this really changes that, other than moving everything up one level. I could perhaps see a use case for specifiying site-scratch with a site tag (so perhaps has_option_tag, or whatever, is better?) but not for doing this as implemented??

Reply:
To be precise, I think this can be done with just *one* option, pycbc|site-scratch, which could be set in `[pegasus_profile]` to affect all sites (which use it), or set on an individual level in `[pegasus_profile-site]` if for e.g. you wanted to use different filesystems for different sites (having condorpool_shared on NFS, but condorpool_copy on a local disk on the headnode could be something sane to do??)

Reply:
@spxiwh Pegasus doesn't ensure they are unique though (yes, they must be or you'll cause problems). It is up to the user to ensure this is this the case. How it's handled now is by using something based on the current directory (i.e. some variation of CWD or similar). One would want this for the site-scratch for basically the same reason as the local-dir. It must be unique, but you don't want to just shunt it to the current working directory. The use case is that you need these scratch location to be under a different filesystem (which is mounted under some path). On ATLAS, for example, we use this to have all scratch read / writes go to a different file server better equipped to properly handle the load (with lots of file caching etc) than standard home file servers.

Reply:
@spxiwh Perhaps the confusion is that I'm talking about wanting to run *multiple workflows*. This not about separating different pools. which indeed already have different names from each other (just not different names than a similar workflow ).

Reply:
Okay, I think I understand the use case now. You give a path to a directory like `/mnt/hfs1/scratch` and want it to make subdirectories on top of that. Perhaps the option name here is confusing? Something like `add_unique_subdir_to_scratch` (I'm rubbish at naming).  

I would still use `has_option_tag` here with the site name as the tag to allow for different directories (presumably different filesystems) for the site scratches if needed.

Then maybe the "unique_subdir" could (should?) be the same for all sites, so that these are logically collected if you are doing the use-case you describe?

Reply:
It is unique per site at the moment. I agree with your latter point, but I'm going to leave for now as unfortunately I need to focus on putting out more pressing fires.